File: tutorials/workflow/csharp/fundamentals/Basic/BasicWorkflow.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Dapr.Workflow;
using Basic.Activities;

namespace Basic;

/// <summary>
/// Workflows inherit from the abstract Workflow class from the Dapr.Workflow namespace.
/// The first generic parameter is the input type of the workflow, and the second parameter is the output type.
/// </summary>
public class BasicWorkflow : Workflow<string, string>
{
    /// <summary>
    /// The RunAsync method is an abstract method in the abstract Workflow class that needs to be implemented.
    /// This method is the entry point for the workflow.
    /// </summary>
    /// <param name="context">The WorkflowContext provides properties about the workflow instance.
    /// It also contains methods to call activities, child workflows, wait for external events, start timers, and continue as a new instance.</param>
    /// <param name="input">The input parameter for the workflow. There can only be one input parameter. Use a record or class if multiple input values are required.</param>
    /// <returns>The return value of the workflow. It can be a simple or complex type. In case a workflow doesn't require an output, you can specify a nullable object since a return type is mandatory.</returns>
    public override async Task<string> RunAsync(WorkflowContext context, string input)
    {
        var result1 = await context.CallActivityAsync<string>(
            nameof(Activity1),
            input);

        var result2 = await context.CallActivityAsync<string>(
            nameof(Activity2),
            result1);

        return result2;
    }
}

File: tutorials/workflow/python/fundamentals/basic/basic_workflow.py

import dapr.ext.workflow as wf

wf_runtime = wf.WorkflowRuntime()

"""
Workflows orchestrate activities and other (child) workflows, and include business logic (if/else or switch statements).
Workflow code must be be deterministic. Any non-deterministic behavior should be written inside activities.

Workflow definitions use the `workflow` decorator to define a workflow.
The first argument (`ctx`) is the `DaprWorkflowContext`, this contains properties
about the workflow instance, and methods to call activities.
The second argument (`wf_input`) is the input to the workflow. It can be a simple or complex type.
"""
@wf_runtime.workflow(name='basic_workflow')
def basic_workflow(ctx: wf.DaprWorkflowContext, wf_input: str):
    result1 = yield ctx.call_activity(activity1, input=wf_input)
    result2 = yield ctx.call_activity(activity2, input=result1)
    return result2

"""
Activity code typically performs a one specific task, like calling an API to store or retrieve data.
You can use other Dapr APIs inside an activity.

Activity definitions use the `activity` decorator to define an activity.
The first argument (`ctx`) is the `WorkflowActivityContext` and provides 
the name of the activity and the workflow instance.
The second argument (`act_input`) is the input parameter for the activity. 
There can only be one input parameter. Use a class if multiple input values are required.
"""
@wf_runtime.activity(name='activity1')
def activity1(ctx: wf.WorkflowActivityContext, act_input: str) -> str:
    print(f'activity1: Received input: {act_input}.', flush=True)
    return f"{act_input} Two"

@wf_runtime.activity(name='activity2')
def activity2(ctx: wf.WorkflowActivityContext, act_input: str) -> str:
    print(f'activity2: Received input: {act_input}.', flush=True)
    return f"{act_input} Three"